src/brod_gssapi_v1.erl

```diff
@@ -149,13 +99,20 @@ sasl_recv(Mod, Sock, Timeout, ClientId, Challenge, HandshakeVsn) ->
 -spec ensure_binary(atom() | iodata()) -> binary().
 ensure_binary(Atom) when is_atom(Atom) ->
     atom_to_binary(Atom, utf8);
-ensure_binary(Str) ->
-    iolist_to_binary(Str).
-
-do_while(Fun, CondFun) ->
-    case CondFun(Fun()) of
+ensure_binary(Str) when is_list(Str) ->
+    iolist_to_binary(Str);
+ensure_binary(Bin) when is_binary(Bin) ->
+    Bin.
+
+maybe_interact_continue(sasl_interact) ->
```

Comment:
This needs to be removed all together. We should never hit `sasl_interact`. I think this was originally based off librdkafka it seems. librdkafka may have a case for interacting with a user (i.e., gui or cli app), but we have none or rather AFAIK we have no way of interacting with the user currently. 

When you provide callbacks to cyrus-sasl for additional information that may needed, you're opting out of getting `sasl_interact` back as a code AFAIK. What's more even if you do get that back at client step, you're supposed to go back to calling client start after you get the information asked for.